### DISCUSSION TITLE
TypeError: Cannot destructure property `control` of 'undefined' or 'null'

### DISCUSSION THREAD

**Discussion by JusJuraev** [react-hook-form/react-hook-form#5105] [Q&A] [answered]
**Describe the bug**
when I try to build my app using nextjs I see this error. My field components are deep nested within Form component and I'm using FormProvider for this situation

**To Reproduce**

1. Go to Sandbox link
2. Open Input.js file
3. Create new terminal
4. run command "yarn build"
5. See error

**Codesandbox link**
https://codesandbox.io/s/friendly-sound-3tqeu

**Expected behavior**
Maybe you should replace createContext(null) with createContext({})?

**Screenshots**
![image](https://user-images.githubusercontent.com/28861774/117581164-e71f9100-b114-11eb-91ec-99ee755fab8f.png)

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser chrome
- Version 7.5

**Comment by bluebill1049** [comment]
This could only be an issue if you are using those inputs without hook form's provider, so double check to see if any inputs are used without the context provider.

**Reply by JusJuraev** [reply]
But this error only occurs at build time, in dev mode everything works perfect. I think when node server runs build, it cannot destructure context value, because of null in default value.

**Reply by bluebill1049** [reply]
i don't think so, the context API is from React Context API. Do you have a GitHub repo to build and reproduce the issue?

**Reply by JusJuraev** [reply]
no, I don't, the project is private, I've created Sandbox. Can you fork and show me correct way please?

**Reply by bluebill1049** [reply]
the one you had in csb looks correct to me.

**Reply by JusJuraev** [reply]
so how can I solve this problem?

**Reply by bluebill1049** [reply]
you will need to investigate your own `input` usage, some input is used without the `context` and hence it failed build.

**Reply by JusJuraev** [reply]
ok, I will. thank you

**Reply by bluebill1049** [reply]
let me know if anything else i can help too. 👍

**Reply by JusJuraev** [reply]
of course 👍

**Reply by JusJuraev** [reply]
bro, can you replace `null `for `{}` for curiosity please? [useFormContext](https://github.com/react-hook-form/react-hook-form/blob/79df5dd7dcdb36a58b06dd36630bcf64012f0558/src/useFormContext.tsx#L6)

I've created a `context` on my own with default value of `null` and recieved same error on build. I'll be very thankful🙏

**Comment by leapful** [comment, marked as answer]
@JusJuraev @bluebill1049 

The current implementation is putting Form, Context, Input components into directory "pages" of Next.js. By default, every component inside "pages" will be pre-compiled at build time to generate static page. Just move your components outside the directory "pages" will fix the issue.

Best practice when working with Next.js is to create a directory "components" to store common and not-page-context components.

**Reply by JusJuraev** [reply]
Thanks a lot, this solution worked for me 🙏🙏🙏
p.s. sorry for late reply :)
